examples/mobile/expo-54-go/components/test-hooks/gas-estimation.tsx

```diff
@@ -0,0 +1,338 @@
+import {
+  useGasEstimation,
+  useSophonAccount,
+} from '@sophon-labs/account-react-native';
+import { useEffect, useMemo, useState } from 'react';
+import { Alert, ScrollView, Text, TextInput, View } from 'react-native';
+import { Button } from '../ui/button';
+
+export const GasEstimation = () => {
+  const { account } = useSophonAccount();
+  const [hasData, setHasData] = useState(false);
+
+  // Form state for config
+  const [formData, setFormData] = useState({
+    to: '0xE676a42fEd98d51336f02510bB5d598893AbfE90', // MOCK MintMe token
+    data: '0xa9059cbb000000000000000000000000feb22da05537b4b63bb63417b10935819facb81c0000000000000000000000000000000000000000000000000de0b6b3a7640000', // transfer(address,uint256)
+    value: '0', // ETH value to send
+    chainId: 531050104, // Sophon Testnet
+    enabled: false,
+  });
+
+  const config = useMemo(
+    () => ({
+      to: formData.to,
+      from: account?.address,
+      data: formData.data,
+      value: BigInt(formData.value),
+      chainId: formData.chainId,
+      enabled: formData.enabled,
+    }),
+    [
+      formData.to,
+      formData.data,
+      formData.value,
+      formData.chainId,
+      formData.enabled,
+      account?.address,
+    ],
+  );
+
+  const { gasEstimate, gasPrice, totalFeeEstimate, isLoading, error, refetch } =
+    useGasEstimation(config);
+
+  // Update hasData when we have meaningful data
+  useEffect(() => {
+    if (gasEstimate !== undefined || gasPrice !== undefined || error) {
+      setHasData(true);
+    }
+  }, [gasEstimate, gasPrice, error]);
+
+  const handleClearData = () => {
+    setHasData(false);
+    setFormData((prev) => ({ ...prev, enabled: false }));
+  };
+
+  const handleEstimate = () => {
+    if (!account?.address) {
+      Alert.alert('Error', 'Please connect your wallet first');
+      return;
+    }
+    // Reset hasData to false when starting new estimation
+    setHasData(false);
+    setFormData((prev) => ({ ...prev, enabled: true }));
+  };
+
+  const handleRefetch = () => {
+    refetch();
+  };
+
+  const handleInputChange = (
+    field: string,
+    value: string | number | boolean,
+  ) => {
+    setFormData((prev) => ({
+      ...prev,
+      [field]: value,
+    }));
+  };
+
+  // Format gas values for display
+  const formatGas = (value: bigint | undefined | null) => {
+    if (!value) return '0';
+    return value.toString();
+  };
+
+  const formatEther = (value: bigint | undefined | null) => {
+    if (!value) return '0';
+    // Convert wei to ether (divide by 10^18)
+    const ether = Number(value) / Number(BigInt(10) ** BigInt(18));
```

Comment:
Converting BigInt to Number may overflow or lose precision for very large values